routers/common/auth.go

```diff
@@ -18,7 +20,8 @@ type AuthResult struct {
 func AuthShared(ctx *context.Base, sessionStore auth_service.SessionStore, authMethod auth_service.Method) (ar AuthResult, err error) {
 	ar.Doer, err = authMethod.Verify(ctx.Req, ctx.Resp, ctx, sessionStore)
 	if err != nil {
-		return ar, err
+		log.Warn("authentication failed", err)
```

Comment:
Maybe we need to detect the errors type to decide whether error we should return.

Reply:
It means to only return ErrBadAccessToken for authentication related errors?

Reply:
The `Verify` method returns various kinds of errors because it includes various auth methods. However, detecting the error type is also a tough task. I can recall there is a bugfix PR trying to fix `500` after deleting a user with the wrong password. But the PR is stuck because it can't cover error types for all cases.

Reply:
@lunny @lng2020 
Maybe this func (`userIDFromToken`) could return type `int64,error` , by modifying the return type, the error type can be returned correctly.


https://github.com/go-gitea/gitea/blob/a18ecaed61365ec8d6688631ab2febd211bd2fe0/services/auth/oauth2.go#L139-L143

If you approve i will close this PR and open another one.